runelite-client/src/main/java/net/runelite/client/plugins/skillcalculator/SkillCalculator.java

```diff
@@ -121,8 +123,9 @@ void openCalculator(CalculatorType calculatorType)
 		// Update internal skill/XP values.
 		currentXP = client.getSkillExperience(calculatorType.getSkill());
 		currentLevel = Experience.getLevelForXp(currentXP);
-		targetLevel = enforceSkillBounds(currentLevel + 1);
-		targetXP = Experience.getXpForLevel(targetLevel);
+		VarPlayer endGoal = endGoalVarpForSkill(calculatorType.getSkill());
+		targetXP = endGoal != null ? client.getVar(endGoal) : -1;
+		targetLevel = enforceSkillBounds(Experience.getLevelForXp(targetXP));
```

Comment:
`Experience.getLevelForXp(-1)` throws an `IllegalArgumentException` and will crash the calculator.